### PR TITLE
fix: ComboBox Single Item Selection Issue

### LINKED ISSUE #2393: Bug with Single-Item Selection with ComboBox
## Current behavior

I am following the instructions in the documentation below to implement selection in a combobox. I discover that combobox is able to identify item firstly selected and the items above it. The items below the firstly selected item are ignored. I see this as a bug because this is not what it is intended to do. 

https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Mvux/Advanced/Selection.html#implement-selection-in-the-peopleapp

## Expected behavior

The expected behaviour should be to identify every selection when it is made.

## How to reproduce it (as minimally and precisely as possible)
[UnoApp1.zip](https://github.com/user-attachments/files/15978464/UnoApp1.zip)

A sample repo is attached here to check the behavior. The ComboBoxPage.xaml is the affected page

## Environment

Nuget Package (s):

Package Version(s):

Affected platform(s):

- [ ] iOS
- [ ] macOS (AppKit)
- [ ] Mac Catalyst
- [ ] Android
- [ ] WebAssembly
- [x] Windows
- [ ] Skia (WPF)
- [x] Skia (GTK on Linux/macOS/Windows)
- [ ] Skia (Linux Framebuffer)
- [ ] Build tasks

Visual Studio:

- [ ] 2019 (version: )
- [x] 2022 (version: )
- [ ] Visual Studio Code (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )

Relevant plugins:

- [ ] Resharper (version: )

## Anything else we need to know?

None. Everything needed is already supplied

### PR DESCRIPTION
**Pull request by vatsashah45**
GitHub Issue (If applicable): closes #2393 

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

ComboBox is able to identify item firstly selected and the items above it. But, the items below the firstly selected item are ignored.

## What is the new behavior?

User can select any item in the ComboBox in any order

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md). (for bug fixes / features) 
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Facets/SelectionFacet.cs [hunk 1]
```diff
@@ -244,7 +244,7 @@ private bool SetCurrent(int index, object? value, bool isCancelable = true, bool
 		{
```
Comment (agneszitte) [inline src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Facets/SelectionFacet.cs:244]: Same for this PR @dr1rrb, do we need additional tests?

File: src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Facets/SelectionFacet.cs [hunk 2]
```diff
@@ -284,6 +284,29 @@ public void SelectRange(ItemIndexRange itemIndexRange)
 		}
 	}
 
+	public void ReplaceRange(ItemIndexRange itemIndexRange)
```
Comment (agneszitte) [inline src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Facets/SelectionFacet.cs:287]:
```suggestion

	/// <inheritdoc />
	public void ReplaceRange(ItemIndexRange itemIndexRange)
```

File: src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Services/ISelectionService.cs [hunk 3]
```diff
@@ -10,4 +10,9 @@ internal interface ISelectionService : ISelectionInfo
 	/// Event raise when any properties of the service has changed
 	/// </summary>
 	event EventHandler StateChanged;
+
+	/// <summary>
+	/// Replace the selected range.
+	/// </summary>
```
Comment (agneszitte) [inline src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Services/ISelectionService.cs:16]:
```suggestion
	/// <summary>
	/// Replace the current selection range with the specified item index range.
	/// </summary>
	/// <param name="itemIndexRange">The item index range to select.</param>
```

**Review by dr1rrb — APPROVED** [review]

File: src/Uno.Extensions.Reactive.UI.Tests/Presentation/Bindings/Collections/Given_BindableCollection_Selection.cs [hunk 4]
```diff
@@ -107,4 +181,17 @@ private async Task<(BindableGiven_BindableCollection_Selection_Model, ListView,
 
 		return (vm, lv, lvItems);
 	}
+
+	private async Task<(BindableGiven_BindableCollection_Selection_Model, ComboBox, ICollectionView, MyItem[])> SetupComboBox()
+	{
+		var vm = new BindableGiven_BindableCollection_Selection_Model();
+		var itemsView = new CollectionViewSource { Source = vm.Items }.View;
+		var cb = new ComboBox { ItemsSource = itemsView };
```
Comment (dr1rrb) [inline src/Uno.Extensions.Reactive.UI.Tests/Presentation/Bindings/Collections/Given_BindableCollection_Selection.cs:189]: We don't need (nor want) a `CollectionViewSource`:

```suggestion
		var cb = new ComboBox { ItemsSource = vm.Items };
```

File: src/Uno.Extensions.Reactive.UI.Tests/Presentation/Bindings/Collections/Given_BindableCollection_Selection.cs [hunk 5]
```diff
@@ -107,4 +181,16 @@ private async Task<(BindableGiven_BindableCollection_Selection_Model, ListView,
...
 	}
+
+	private async Task<(BindableGiven_BindableCollection_Selection_Model, ComboBox, ICollectionView, MyItem[])> SetupComboBox()
+	{
+		var vm = new BindableGiven_BindableCollection_Selection_Model();
+		var cb = new ComboBox { ItemsSource = vm.Items };
+
+		await UIHelper.Load(cb, CT);
+
+		var items = vm.Items.Value.ToArray();
```
Comment (dr1rrb) [inline src/Uno.Extensions.Reactive.UI.Tests/Presentation/Bindings/Collections/Given_BindableCollection_Selection.cs:192]: The `.Value` doesn't not exists on an `IFeed` ... copilot is just an helper, don't trust it 😊 Anyway you don't need that, you're not using them in your test methods. 

Note: You als have to remove the `ICollectionView` from the return tuple type of your method ;)

File: src/Uno.Extensions.Reactive.UI.Tests/Presentation/Bindings/Collections/Given_BindableCollection_Selection.cs [hunk 6]
```diff
@@ -107,4 +181,16 @@ private async Task<(BindableGiven_BindableCollection_Selection_Model, ListView,
...
 	}
+
+	private async Task<(BindableGiven_BindableCollection_Selection_Model, ComboBox, MyItem[])> SetupComboBox()
+	{
+		var vm = new BindableGiven_BindableCollection_Selection_Model();
+		var cb = new ComboBox { ItemsSource = vm.Items };
+
+		await UIHelper.Load(cb, CT);
+
+		var items = vm.Items.ToArray();
```
Comment (dr1rrb) [inline src/Uno.Extensions.Reactive.UI.Tests/Presentation/Bindings/Collections/Given_BindableCollection_Selection.cs:192]: This `ToArray()` does not exists, please remove the `items` as you are not using them anyway.

**Comment by agneszitte** [comment]
@dr1rrb and @kazo0 can you do a last review please

**Review by dr1rrb — APPROVED** [review]

**Pull request merged**